File: software/examples/python/example_threshold.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_industrial_dual_0_20ma import IndustrialDual020mA

# Callback for current greater than 10mA
def cb_reached(sensor, current):
    print('Current (Sensor' + str(sensor) + ') is greater than 10mA: ' + str(current/1000000.0))

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    id020 = IndustrialDual020mA(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get threshold callbacks with a debounce time of 10 seconds (10000ms)
    id020.set_debounce_period(10000)

    # Register threshold reached callback to function cb_reached
    id020.register_callback(dual020.CALLBACK_CURRENT_REACHED, cb_reached)

    # Configure threshold (sensor 1) for "greater than 10mA" (unit is nA)
    id020.set_current_callback_threshold(1, '>', 10*1000000, 0)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/javascript/ExampleThreshold.js
var IPConnection = require('Tinkerforge/IPConnection');
var BrickletIndustrialDual020mA = require('Tinkerforge/BrickletIndustrialDual020mA');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'ftn';// Change to your UID

var ipcon = new IPConnection();// Create IP connection
var dual020 = new BrickletIndustrialDual020mA(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        dual020.setDebouncePeriod(10000);
        // Configure threshold (sensor 1) for "greater than 10mA" (unit is nA)
        dual020.setCurrentCallbackThreshold(1, '>', 10*1000*1000, 0);      
    }
);

// Register threshold reached callback
dual020.on(BrickletIndustrialDual020mA.CALLBACK_CURRENT_REACHED,
    // Callback for current greater than 10mA
    function(sensor, current) {
        console.log('Current (sensor '+sensor+') is greater than 10mA: '+current/(1000*1000)+' mA');
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

